samples/core/tutorials/estimators/wide.ipynb

```diff
@@ -0,0 +1,1273 @@
+      ]
+    },
+    {
+      "metadata": {
+        "id": "QZZtXes4cYvf",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "The columns can be grouped into two types—categorical\n",
+        "and continuous columns:\n",
+        "\n",
+        "*   A column is called **categorical** if its value can only be one of the\n",
+        "    categories in a finite set. For example, the relationship status of a person\n",
+        "    (wife, husband, unmarried, etc.) or the education level (high school,\n",
+        "    college, etc.) are categorical columns.\n",
+        "*   A column is called **continuous** if its value can be any numerical value in\n",
+        "    a continuous range. For example, the capital gain of a person (e.g. $14,084)\n",
+        "    is a continuous column.\n",
+        "\n",
+        "Here's a list of columns available in the Census Income dataset:\n",
+        "\n",
+        "## Converting Data into Tensors\n",
+        "\n",
+        "When building a tf.estimator model, the input data is specified by means of an\n",
+        "input function or `input_fn`. This builder function returns a `tf.data.Dataset`\n",
+        "of batches of `(features-dict,label)` pairs. It will not be called until it is\n",
+        "later passed to `tf.estimator.Estimator` methods such as `train` and `evaluate`.\n",
+        "\n",
+        "In more detail, the input builder function returns the following as a pair:\n",
+        "\n",
+        "1.  `features`: A dict from feature names to `Tensors` or\n",
+        "    `SparseTensors` containing batches of features.\n",
+        "2.  `labels`: A `Tensor` containing batches of labels.\n",
+        "\n",
+        "The keys of the `features` will be used to configure the model's input layer.\n",
+        "\n",
+        "Note that the input function will be called while\n",
+        "constructing the TensorFlow graph, not while running the graph. What it is\n",
+        "returning is a representation of the input data as sequence of tensorflow graph\n",
+        "operations.\n",
+        "\n",
+        "For small problems like this it's easy to make a `tf.data.Dataset` by slicing the `pandas.DataFrame`:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "N7zNJflKcYvg",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def easy_input_function(df, label_key, num_epochs, shuffle, batch_size):\n",
+        "  df = df.copy()\n",
+        "  label = df.pop(label_key)\n",
```

Comment:
This may be my mis-memory of Pandas, which I admit I haven't used in a while, but do we have to create copies of the dataframe? Can't we just say, `label = df[label_key]` or something to that effect?

Reply:
> label = df[label_key]

That works. Fixed.
 The only down side is that the label is still in the features dict.